heareval/predictions/task_predictions.py

```diff
@@ -146,38 +286,64 @@ def __init__(
         self.labels = pickle.load(
             open(embedding_path.joinpath(f"{split_name}.target-labels.pkl"), "rb")
         )
+        # Only used for event-based prediction
+        # For timestamp (event) embedding tasks,
+        # the metadata for each instance is {filename: , timestamp: }.
+        if self.embedding_type == "event":
+            filename_timestamps_json = embedding_path.joinpath(
+                f"{split_name}.filename-timestamps.json"
+            )
+            self.metadata = [
+                {"filename": filename, "timestamp": timestamp}
+                for filename, timestamp in json.load(open(filename_timestamps_json))
+            ]
+        else:
+            self.metadata = [{}] * self.dim[0]
         assert len(self.labels) == self.dim[0]
-        assert (
-            self.embedding_memmap[0].shape[0] == self.dim[1]
-        ), f"{self.embedding_memmap[0].shape[0]}, {self.dim[1]}"
+        assert len(self.labels) == len(self.embedding_memmap)
+        assert len(self.labels) == len(self.metadata)
+        assert self.embedding_memmap[0].shape[0] == self.dim[1]
 
     def __len__(self) -> int:
         return self.dim[0]
 
-    def __getitem__(self, idx) -> Tuple[torch.Tensor, torch.Tensor]:
+    def __getitem__(self, idx) -> Tuple[torch.Tensor, torch.Tensor, Dict[str, Any]]:
         """
         For all labels, return a multi or one-hot vector.
         This allows us to have tensors that are all the same shape.
         Later we reduce this with an argmax to get the vocabulary indices.
+            We also include the filename and timestamp, which we need
+        for evaluation of timestamp (event) tasks.
+        We also return the metadata as a Dict.
         """
         x = self.embedding_memmap[idx]
         y = [self.label_to_idx[str(label)] for label in self.labels[idx]]
         # Lame special case
         if not y:
-            return x, torch.zeros((self.nlabels,), dtype=torch.int32)
+            return (
+                x,
+                # BCEWithLogitsLoss wants float not long targets
+                torch.zeros((self.nlabels,), dtype=torch.int32).float(),
+                self.metadata[idx],
+            )
         # TODO: Could rewrite faster using scatter_:
         # https://discuss.pytorch.org/t/what-kind-of-loss-is-better-to-use-in-multilabel-classification/32203/4
         return (
             x,
+            # BCEWithLogitsLoss wants float not long targets
             torch.nn.functional.one_hot(torch.LongTensor(y), num_classes=self.nlabels)
             .max(axis=0)
-            .values,
+            .values.float(),
+            self.metadata[idx],
         )
 
 
 def create_events_from_prediction(
     prediction_dict: Dict[float, torch.Tensor],
```

Comment:
Yeah so this params shouldn't be hardcoded in predict really, it should be a part of the event scoring classes. I am not really sure how to move these hyperparams to make them in scores.py but that's where they belong.

Also these methods are really kind confusing here, I think they belong somewhere else.

Reply:
Could we move these into the score function somehow? Like a sed eval function that receives a set of frame embedding predictions / targets and then creates the events from those and calculates the scores? So we don't need to fuss with the conversion beforehand?

Reply:
We might want to use this to dump the events in the test part (I see commented code for the test). But putting in score also sounds like a good option